pkg/archive/archive_cgo.go

```diff
@@ -0,0 +1,18 @@
+// +build cgo
+
+package archive
+
+import (
+	"bufio"
+	"io"
+
+	"github.com/DataDog/zstd"
+)
+
+func zstdReader(buf *bufio.Reader) (io.ReadCloser, error) {
```

Comment:
Why does this expect a `*bufio.Reader` instead of an `io.Reader`?

Reply:
I had left that as with unbuffered input I see very bad compression results.

But I guess it is better to be more generic, I've changed it to `io.Reader`.